### PR TITLE
[HYDRATOR-1124] Fixed record length calculations in cobol reader plugin

### PR DESCRIPTION
**Pull request by romy-khetan**

### CODE REVIEW

File: copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookIOUtils.java [hunk 1]
```diff
@@ -61,8 +63,12 @@ public static ExternalRecord getExternalRecord(InputStream cblIs) throws RecordE
    */
   public static int getRecordLength(ExternalRecord externalRecord, int fileStructure) {
     int recordByteLength = 0;
+    Set<Integer> fieldPositions = new HashSet<>();
     for (ExternalField field : externalRecord.getRecordFields()) {
-      recordByteLength += field.getLen();
+      if (!fieldPositions.contains(field.getPos())) {
+        recordByteLength += field.getLen();
+        fieldPositions.add(field.getPos());
+      }
```
Comment (abhinavbansal246) [inline copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookIOUtils.java:71]: This code is not accounting field.getPos() while calculating  recordByteLength. The suggested code was
```
  public static int getRecordLength(ExternalRecord externalRecord, int fileStructure) {
    int recordByteLength = 0;
    for (ExternalField field : externalRecord.getRecordFields()) {
      recordByteLength = Math.Max(recordByteLength, field.getPos() + field.getLen());
    }
    return recordByteLength;
  ```

Comment (romy-khetan) [reply copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookIOUtils.java:71]: Analysed the suggested fix but it calculates the record length incorrectly. And, it seems the record length in combination with the splits, gets the parsing of Cobol records done in a wrong manner and throws exceptions. 

Therefore, considering the use case suggested, where the existing length calculation fails, I have applied another approach to fix the issue.

**Comment by abhinavbansal246** [comment]
LGTM

**Comment by romy-khetan** [comment]
Pasting ITN results...
[INFO] --- maven-checkstyle-plugin:2.12.1:check (validate) @ long-running-test ---
[INFO]
[INFO] --- maven-surefire-plugin:2.18.1:test (default-test) @ long-running-test ---
[INFO] Tests are skipped.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] integration-test ................................... SUCCESS [  2.504 s]
[INFO] integration-test-core .............................. SUCCESS [  0.526 s]
[INFO] integration-test-remote ............................ SUCCESS [04:45 min]
[INFO] integration-test-standalone ........................ SUCCESS [  0.390 s]
[INFO] upgrade-test ....................................... SUCCESS [  0.244 s]
[INFO] long-running-test .................................. SUCCESS [  0.304 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:49 min
[INFO] Finished at: 2016-12-06T19:01:35+05:30
[INFO] Final Memory: 31M/201M
[INFO] ------------------------------------------------------------------------

**Pull request merged**